packages/core/core-flows/src/product/helpers/normalize-for-import.ts

```diff
@@ -108,10 +116,15 @@ const normalizeProductForImport = (
     }
 
     if (normalizedKey.startsWith("product_tag_")) {
-      response["tags"] = [
-        ...(response["tags"] || []),
-        { value: normalizedValue },
-      ]
+      const tag = tagsMap.get(normalizedValue)
+      if (!tag) {
```

Comment:
I would be ok for the import to first create all tags and then create the products if we want that behavior (as a separate PR though).